user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+# Add your Custom Data Sources with Plugin Framework
+
+Exploratory provides a framework with which developers can create their own data sources that can be invoked from Exploratory's Data Import Dialog.
+
+## Plugin Structure
+
+A User defined plugin consists of JSON format plugin definition meta file (`plugin.json`) and R script file (`plugin.r`).
+These two files need to be stored inside `~/.exploratory/plugins` directory as follows.
+
+`~/.exploratory/plugins/<plugin_name>/plugin.json`
+
+`~/.exploratory/plugins/<plugin_name>/scripts/plugin.r`
+ 
+
+## Create Plugin directory
+
+So let's say you want to create a plugin that uses `riem_measures` function from [riem](https://github.com/ropensci/riem) R package to get ASOS data via the Iowa Environment Mesonet. Then first you want to to create a directory called `riem_measures`. NOTE: The directory name is the plugin name and should be unique among user-defined plugins.
+
+So on Mac, you can do:
+
+`$ cd ~/.exploratory/plugins`
+
+`$ mkdir riem_measures`
+
+## Create Plugin Definition Meta File.
+
+This is how your Plugin Definition Meta File (`plugin.json`) looks like. The file must be in valid JSON format.
+
+```
+{
+  "name": "riem_measures",
+  "displayName": "riem measures",
+  "iconURL" : "lib/images/plugin.png",
+  "helpURL" :  "http://ropensci.github.io/riem/",
+  "rPackageDependencies": ["riem"],
+  "function": "riem_measures",
+  "rSourceFile" : "plugin.r",
+  "hasQueryField" : false,
+  "inputParameters": [
+    {
+      "name": "station",
+      "displayName":"Station",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    },
+    {
+      "name": "date_start",
+      "displayName":"Date Start",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    },
+    {
+      "name": "date_end",
+      "displayName":"Date End",
+      "colSpan" : 5,
+      "dataType": "text",
+      "defaultValue" : "",
+      "required" : true
+    }
+  ]
+}
+```
+
+### Attributes
+
+#### name (required)
+
+`name` attribute holds Name of the plugin. in this case `riem_measurs`, please make sure to use your directory name for this `name` attribute. 
+
+#### displayName (required)
+
+`displayName` attribute is used for Display Name on Data Source Picker Dialog and Import Data Dialog.
+
+#### iconURL (required)
+
+`iconURL` holds icon image file URL for your plugin. currently it only supports external URL.
+
+For example:
+ 
+```
+   "iconURL" : "http://xmllondon.com/images/sparqlThumb.png",
+```
+
+If you do not have an icon, please use below for default lego like block icon.
```

Comment:
Shouldn't we automatically show this when iconURL is not set without having user to specify it?